core/app/models/concerns/spree/user_address_book.rb

```diff
@@ -0,0 +1,73 @@
+module Spree
+  module UserAddressBook
+    extend ActiveSupport::Concern
+
+    included do
+      has_many :user_addresses, foreign_key: "user_id", class_name: "Spree::UserAddress"
+
+      has_many :addresses, through: :user_addresses
+
+      has_one :default_user_address, -> { where default: true}, foreign_key: "user_id", class_name: 'Spree::UserAddress'
+      has_one :default_address, through: :default_user_address, source: :address
+
+      # bill_address is only minimally used now, but we can't get rid of it without a major version release
```

Comment:
I have my vision for the long term, but ultimately it will depend on the support of the core team. I think of the UserAddressBook as a list of addresses for easy pre-fill no matter whether the destination is a bill or ship address. User#bill_address _could_ become as simple as `default_credit_card.address`. Based upon that, I (personally) see user.bill_address as potentially going away, or else being an opt-in rather than the default.